### PR TITLE
IGNITE-15224 Thin 3.0: Implement Table API operations

### PR DESCRIPTION
**Pull request by ptupitsyn**
* Implement all Table API operations except invoke and transactions.
* Override packer/unpacker methods without checked exceptions for cleaner async code.

### CODE REVIEW

File: modules/client-common/src/main/java/org/apache/ignite/client/proto/ClientMessagePacker.java [hunk 1]
```diff
@@ -60,29 +63,250 @@ public ClientMessagePacker(ByteBuf buf) {
...
+
     /**
      * Writes an UUID.
      *
      * @param val UUID value.
      * @return This instance.
-     * @throws IOException when underlying output throws IOException.
      */
-    public ClientMessagePacker packUuid(UUID val) throws IOException {
+    public ClientMessagePacker packUuid(UUID val) {
```
Comment (isapego) [inline modules/client-common/src/main/java/org/apache/ignite/client/proto/ClientMessagePacker.java:307]: This function seems like it's going to have bad impact on performance. Allocating new array and wrapping it in ByteBuffer every time we pack a UUID seems wrong to me. Maybe it should be handled in separate ticket though,

**Review by isapego — APPROVED** [review]
Overall looks good

File: modules/client-common/src/main/java/org/apache/ignite/client/proto/ClientMessagePacker.java (hunk 1 above)
Comment (ptupitsyn) [reply modules/client-common/src/main/java/org/apache/ignite/client/proto/ClientMessagePacker.java:307]: Good point. Added TODO with a link to https://issues.apache.org/jira/browse/IGNITE-15234.
We'll be able to write two longs directly to the underlying `ByteBuf` with those changes.

Comment (isapego) [reply modules/client-common/src/main/java/org/apache/ignite/client/proto/ClientMessagePacker.java:307]: Good.

**Pull request merged**